src/sql/workbench/contrib/query/browser/query.contribution.ts

```diff
@@ -227,6 +228,25 @@ actionRegistry.registerWorkbenchAction(
 	'Change Language Flavor'
 );
 
+// Query Result writer actions
+actionRegistry.registerWorkbenchAction(
+	SyncActionDescriptor.create(
+		QueryResultsToGridAction,
+		QueryResultsToGridAction.ID,
+		QueryResultsToGridAction.LABEL
+	),
+	'Query Results to Grid'
```

Comment:
I think it'd be better to have a single action to change the output mode - and then the user selects the mode they want from a list (like how language selection for files works)

Although have we thought about letting users output to multiple places? There doesn't seem to be any reason why a user couldn't output to both the messages pane and a file.

Reply:
I hadn't considered multiple output modes since I was aiming to match the output behavior in SSMS, but adding that should be relatively simple.

The implementation for that would consist of having an array of writers. Each of those writers would then need to be registered to the query runner callbacks, and then the writers would handle formatting and writing.